hokusai/lib/environment.py

```diff
@@ -0,0 +1,27 @@
+import os
+import sys
+
+import hokusai
+
+# detect whether code is run from a PyInstaller-created bundle
+# if is, support files are found in paths as coded in PyInstaller spec file (see 'datas' arg of Analysis() call)
+def frozen():
+  return getattr(sys, 'frozen', False)
+
+def cert_file_path():
+  if frozen():
+    return sys._MEIPASS + '/lib/cert.pem'
+  else:
+    return ''
+
+def templates_dir_path():
+  if frozen():
+    return sys._MEIPASS + '/hokusai_datas/templates'
+  else:
+    return os.path.dirname(hokusai.__file__) + '/templates'
+
+def version_file_path():
+  if frozen():
+    return sys._MEIPASS + '/hokusai_datas/VERSION'
+  else:
+    return os.path.dirname(hokusai.__file__) + '/VERSION'
```

Comment:
> wonder if this lib is in good practice

Could you clarify what you mean? If regarding file location, seems good to me.

Reply:
whether this way of detecting runtime environment is good practice. i fear it might not be portable across different local environments (MacOS, Linux, Docker). but if we don't see a better way then it will have to do.